oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java

```diff
@@ -131,9 +144,49 @@ public static Builder newBuilder(IdentityPoolCredentials identityPoolCredentials
     return new Builder(identityPoolCredentials);
   }
 
+  private IdentityPoolSubjectTokenSupplier createCertificateSubjectTokenSupplier(
+      Builder builder, IdentityPoolCredentialSource credentialSource) {
+    try {
+      // Configure the mTLS transport with the x509 keystore.
+      X509Provider x509Provider = getX509Provider(builder, credentialSource);
+      KeyStore mtlsKeyStore = x509Provider.getKeyStore();
+      this.transportFactory = new MtlsHttpTransportFactory(mtlsKeyStore);
+
+      // Initialize the subject token supplier with the certificate path.
+      credentialSource.setCredentialLocation(x509Provider.getCertificatePath());
+      return new CertificateIdentityPoolSubjectTokenSupplier(credentialSource);
+
+    } catch (IOException e) {
+      throw new RuntimeException(
+          // Wrap IOException in RuntimeException because constructors cannot throw checked
+          // exceptions.
+          "Failed to initialize IdentityPoolCredentials from certificate source due to an I/O error.",
```

Comment:
nit: It's a bit strange to have this method do this given this is a separate method, even if it is only called by the constructor

Reply:
Moved the try/catch block to the constructor.